RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,174 @@
+
+We'll need the `quant.sf` files for all the samples in an experiment.
+
+## Libraries and functions
+
+```{r library, live = TRUE}
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+```{r directories, live = TRUE}
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path("data", "gastric-cancer", "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path("data", "gastric-cancer", "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+**Output**
+
+```{r output-names, live = TRUE}
+# Name the output gastric-cancer_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "gastric-cancer_tximport.RDS")
+```
+
+### Naming
+
+All output files from `salmon quant` we'll use with `tximport` are named 
+`quant.sf`.
+The file names do not have any information about the sample name.
+
+```{r sf_files, live = TRUE}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` 
+package.
+
+Notice how the file path is separated by `/`.
+If we were to split up this character by `/`, the fourth item is the sample 
+name.
+This is exactly what `stringr::word` allows us to do: split up the file paths
+by `/` and extract the sample names.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, 4, sep = "/")
+sample_names
+```
+
+`tximeta` needs a data.frame with these columns: 
+- a `files` column  with the file paths to the quant.sf files
+- a `names` column with the sample names
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names, 
+                      stringsAsFactors = FALSE)
+```
+
+## Set up our tximeta object
+
+Using our `coldata` data frame we set up, we can run the `tximeta()` which will 
+extract species specific information.
+
+```{r read_tsv, live = TRUE}
+txi_meta <- tximeta(coldata)
+```
+
+*There are extra steps needed if you are using any species other than human or mouse.
+See our other species examples in `tximeta_for_other_species.Rmd`. 
+
+### Summarize to gene
+
+We'll summarize to the gene level using the `summarizeToGene()` function.
+
+```{r make-txi}
+# Summarize to the gene-level
+gene_summarized <- summarizeToGene(txi_meta) 
+```
+
+We can use the `class` function to see what type of object `gene_summarized` is.
+
+```{r class-txi, live = TRUE}
+# Check what type of object gene_summarized is
```

Comment:
```suggestion
# Check what type of object `gene_summarized` is
```

Reply:
I think we'll still want the backticks around object names in comments?

Reply:
I think so.